include/CL/sycl/buffer/detail/buffer.hpp

```diff
@@ -157,27 +186,47 @@ class buffer : public detail::buffer_base,
          event available_event)
   */
 
+
   /** The buffer content may be copied back on destruction to some
       final location */
   ~buffer() {
-    /* If there is a final_data set and that points to something
-       alive, copy back the data through the shared pointer */
-    if (auto p = final_data.lock())
-      std::copy_n(access.data(), access.num_elements(), p.get());
-    /* If data are shared with the host but not concretely, we would
-       have to copy back the data to the host */
-    // else if (shared_data)
-    //   std::copy_n(access.data(), access.num_elements(), shared_data.get());
+    if (modified && final_write_back)
+      (*final_write_back)();
   }
 
+
+  /** Enforce the buffer to be considered as being modified.
+      Same as creating an accessor with write access.
+   */
+  void mark_as_written() {
+    modified = true;
+  }
+
+
   // Use BOOST_DISABLE_ASSERTS at some time to disable range checking
 
-  /// Return an accessor of the required mode \param M
-  /// \todo Remove if not used
+
+  /** This method is to be called whenever an acessor is created.
+      Its current purpose is to track if an accessor with write access
+      is created and acting acordingly.
+   */
   template <access::mode Mode,
-            access::target Target = access::target::global_buffer>
-  detail::accessor<T, Dimensions, Mode, Target> get_access() {
-    return { *this };
+            access::target Target = access::target::host_buffer>
+  void track_access_mode() {
+    if (   Mode == access::mode::write
+        || Mode == access::mode::read_write
+        || Mode == access::mode::discard_write
+        || Mode == access::mode::discard_read_write
+        || Mode == access::mode::atomic
+       ) {
+      modified = true;
```

Comment:
Comment the code up to the end of the function so a reader can understand the idea behind.